omod-2.0/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/converter/openmrs2_0/AllergenConverter2_0.java

```diff
@@ -0,0 +1,100 @@
+/**
+ * The contents of this file are subject to the OpenMRS Public License
+ * Version 1.0 (the "License"); you may not use this file except in
+ * compliance with the License. You may obtain a copy of the License at
+ * http://license.openmrs.org
+ *
+ * Software distributed under the License is distributed on an "AS IS"
+ * basis, WITHOUT WARRANTY OF ANY KIND, either express or implied. See the
+ * License for the specific language governing rights and limitations
+ * under the License.
+ *
+ * Copyright (C) OpenMRS, LLC.  All Rights Reserved.
+ */
+package org.openmrs.module.webservices.rest.web.v1_0.converter.openmrs2_0;
+
+import org.openmrs.Allergen;
+import org.openmrs.AllergenType;
+import org.openmrs.Concept;
+import org.openmrs.annotation.Handler;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.webservices.rest.SimpleObject;
+import org.openmrs.module.webservices.rest.web.RestConstants;
+import org.openmrs.module.webservices.rest.web.api.RestService;
+import org.openmrs.module.webservices.rest.web.representation.Representation;
+import org.openmrs.module.webservices.rest.web.representation.DefaultRepresentation;
+import org.openmrs.module.webservices.rest.web.resource.api.Converter;
+import org.openmrs.module.webservices.rest.web.response.ConversionException;
+import org.openmrs.module.webservices.rest.web.v1_0.resource.openmrs1_11.ConceptResource1_11;
+import org.apache.commons.lang3.StringUtils;
+import java.util.LinkedHashMap;
+
+/**
+ * An implementation of Converter to be able to create a representation from a User
+ * when User is used in another resource. Currently UserAndPassword doesn't convert User
+ */
+@Handler(supports = Allergen.class, order = 0)
+public class AllergenConverter2_0 implements Converter<Allergen> {
```

Comment:
If you extend org.openmrs.module.webservices.rest.web.resource.impl.BaseDelegatingConverter then that will probably simplify your code here. E.g. you don't need to implement getProperty/setProperty